test/model/game_state_test.dart

```diff
@@ -115,5 +117,48 @@ void main() {
     });
   });
 
+  group('Special cards', ()
+  {
+    test('should put Mutiny in next if too many resources', () {
+      const plunderAWreck = PlunderAWreck();
+      const aRivalShip = ARivalShip();
+      const mutiny = Mutiny();
+      final GameState state = makeGameState(
+          deck: [plunderAWreck, aRivalShip],
+          playerHand: [Resource.DOUBLOON, Resource.DOUBLOON, Resource.DOUBLOON, Resource.DOUBLOON,
+          Resource.DOUBLOON, Resource.DOUBLOON, Resource.DOUBLOON, Resource.DOUBLOON,
+          Resource.DOUBLOON, Resource.DOUBLOON, Resource.DOUBLOON, Resource.DOUBLOON,
+          Resource.DOUBLOON, Resource.DOUBLOON, Resource.DOUBLOON, Resource.DOUBLOON]
+      );
+
+      expect(state.nextCard(), mutiny);
+    });
+
+    test('should put Navy Raid in next if too much infamy', () {
+      const plunderAWreck = PlunderAWreck();
+      const aRivalShip = ARivalShip();
+      final GameState state = makeGameState(
+          deck: [plunderAWreck, aRivalShip],
+          playerHand: [Resource.INFAMY, Resource.INFAMY, Resource.INFAMY, Resource.INFAMY,
+            Resource.INFAMY, Resource.INFAMY, Resource.DOUBLOON, Resource.DOUBLOON,
+            Resource.DOUBLOON, Resource.DOUBLOON, Resource.DOUBLOON, Resource.DOUBLOON]
+      );
+      // TODO - currently it's reliant on toss of a coin
+      expect(state.nextCard(), [NavyRaid]);
+    });
+
+    test('should put Ravenous Crew in next if no food', () {
+      const plunderAWreck = PlunderAWreck();
+      const aRivalShip = ARivalShip();
+      final GameState state = makeGameState(
+          deck: [plunderAWreck, aRivalShip],
+          playerHand: [Resource.DOUBLOON, Resource.DOUBLOON, Resource.DOUBLOON, Resource.DOUBLOON,
+            Resource.DOUBLOON, Resource.DOUBLOON, Resource.DOUBLOON, Resource.DOUBLOON,
+            Resource.DOUBLOON, Resource.DOUBLOON, Resource.DOUBLOON, Resource.DOUBLOON]
+      );
+
+      expect(state.nextCard(), [RavenousCrew]);
```

Comment:
Similarly this can become: `expect(state.nextCard(), new TypeMatcher<RavenousCrew>());`

Reply:
Agreed, done.